.gitpod.yml

```diff
@@ -13,16 +13,13 @@ ports:
 
 tasks:
   - name: Forem Server
-    before: |
-      redis-server &
-    init: >
-      gitpod-init.sh
+    before: redis-server &
+    init: ./gitpod-init.sh
+    command: bin/startup
```

Comment:
Based on my "longer-form" message above I think this might work on a fresh Gitpod install (first-time Gitpod user/account).

```suggestion
    command: bin/setup && bin/startup
```

Reply:
I actually do this all the time when switching branches. Yes it takes a minute to run and sometimes you don't need to pull all recent gems, but it ensures the environment matches the Gemfile and node modules.

Reply:
`bin/setup` is run in the Gitpod init script.

https://github.com/forem/forem/blob/main/gitpod-init.sh#L11

The `awaiting port 3000...` message is running in one terminal session and in the `Forem Server` terminal session the `./gitpod-init.sh` is running. It takes a few minutes the first time even when Gitpod's ed itor UI loads up. We can probably rework the `awaiting port 3000...` message and flow to make it more apparent that things are still installing.

Reply:
In regards to the gitpod URL in https://github.com/forem/forem/blob/main/config/environments/development.rb#L80-L83, we can probably scrap that as we already handle it in `./gitpod-init.sh like so:

```
cp .env_sample .env
echo "APP_DOMAIN=\"$(gp url 3000 | cut -f 3 -d /)\"" >> .env
echo 'APP_PROTOCOL="https://"' >> .env
```

so I don't think that's the issue. I haven't tried to log in yet though as I was just concerned about getting the environment up. I can definitely take a peek at that though.

Reply:
> I installed the missing plugins and realized there's two terminal tabs on the lower right hand side: Forem Server: bash and Open site: bash. The Forem Server terminal had crashed and re-running bin/startup wouldn't work.

The plugins and other toast messages are just suggesting to install recommended VS Code plugins (this happens as well if your in VS Code on your local). The other message is saying 3 plugins don't work in a VS Code browser based environment, and the last message is just asking about fetching latest code occasionally. All of that could probably be echoed in the terminal to explain that to the developer so as to not confuse them if they are not used to a VS Code like experience.

![image](https://user-images.githubusercontent.com/833231/133865063-472e74a3-a20a-475e-b5a9-8d8bd5793770.png)